### DISCUSSION TITLE
[QST] TMA copy predicate

### DISCUSSION THREAD

**Discussion by hyhieu** [NVIDIA/cutlass#1344] [Q&A] [answered]
Hi everyone,

I have a conceptual question: in `CuTe`, is it possible to do predicate copy with TMA ops such as `SM90_TMA_LOAD`? If yes, what is the approach? Perhaps something like `copy_if()`.

Thanks!

**Comment by thakkarV** [comment, marked as answer]
TMA itself supports predication. If you pass in a base coordinate whose tile lies outside of the tensor shape in the TMA descriptor, those OOB loads will be predicated off by TMA itself.
